zbus/src/proxy/mod.rs

```diff
@@ -358,6 +358,12 @@ impl PropertiesCache {
                 Some(Either::Left(_update)) => {
                     // discard updates prior to the initial population
                 }
+                Some(Either::Right(Err(zbus::Error::MethodError(name, _, _))))
```

Comment:
Interesting. So we don't end up with a `zbus::fd::Error::ServiceUnknown`? :cry:

Reply:
I guess that's because this is using `call_method_raw` to call `GetAll`. So it just gets a `zbus::Error` instead of a `zbus::fdo::Error`.

Reply:
> We can call `zbus::fdo::Error::from()` here.

That'd be better but I want to solve it more generally so one gets the more specific error and here you can then just match on that. Hence why I laid out the plan I did in the previous comment.

Reply:
I'm not sure exactly what you're suggesting.

If we convert to a `zbus::fdo::Error`, that will consume the `zbus::Error`; so we need to clone or convert back if we want to return the original error. I think we want to just ignore `ServiceUnknown` here, but still return an error for other `zbus::fdo::Error` variants, (both the `zbus::fdo::Error::Zbus` fallthrough case, and the others).

Reply:
> so we need to clone or convert back if we want to return the original error.

As I wrote:

> From<Message> impl should convert the Error into an fdo::Error and wrap it in Error::Fdo, unless the conversion ends up with the fall-through fdo::Error::ZBus variant, in which case we discard the fdo::Error and keep the original Error.

Reply:
> So something like this?

I think so, yes.

> (Looks a little funny converting a `zbus::Error` to an `fdo::Error` and back like this

True true but maybe we can add some comments and make the types explicit to make it easy to read the code.

> but `From<fdo::Error> for Error` already does a match on whether its an `fdo::Error::Zbus` or not, and otherwise creates a `Error::FDO`)

Ah right, true.